scripts/custom/clean_dateTime.py

```diff
@@ -0,0 +1,60 @@
+import datetime
+
+from scripts import utils
+
+
+def clean_dateTime(raw_input):  # noqa: C901
+    if not isinstance(raw_input, str):
+        raw_input = str(raw_input)
+    if utils.is_empty(raw_input):
+        return ""
+
+    date = None
+
+    # Handle YYYY
+    try:
+        date = datetime.datetime.strptime(raw_input, "%Y")
+        result = date.strftime("%Y")
+    except ValueError:
+        pass
+
+    # Handle YYYYMM
+    try:
+        date = datetime.datetime.strptime(raw_input, "%Y%m")
+        result = date.strftime("%Y%m")
+    except ValueError:
+        pass
+
+    # Handle YYYYMMDD
+    try:
+        date = datetime.datetime.strptime(raw_input, "%Y%m%d")
+        result = date.strftime("%Y-%m-%d")
+    except ValueError:
+        pass
+
+    # Handle YYYY-MM-DDTH:M:S
+    try:
+        date = datetime.datetime.strptime(raw_input, "%Y-%m-%dT%H:%M:%S")
+        result = date.strftime("%Y%m%dT%H:%M:%S")
+    except ValueError:
+        pass
+
+    # Handle YYYY-MM-DD H:M:S
+    try:
+        date = datetime.datetime.strptime(raw_input, "%Y-%m-%d %H:%M:%S")
+        result = date.strftime("%Y-%m-%d %H:%M:%S")
```

Comment:
This is not the right format to return, you need the T between date and time see : https://www.hl7.org/fhir/datatypes.html#dateTime